src/usr/local/www/system_certmanager.php

```diff
@@ -335,17 +345,29 @@
 
 			switch ($pconfig['method']) {
 				case "internal":
+					if (isset($_POST["keytype"]) && !in_array($_POST["keytype"], $cert_keytypes)) {
+						array_push($input_errors, gettext("Please select a valid Key Type."));
+					}
 					if (isset($_POST["keylen"]) && !in_array($_POST["keylen"], $cert_keylens)) {
 						array_push($input_errors, gettext("Please select a valid Key Length."));
 					}
+					if (isset($_POST["eckey"]) && !in_array($_POST["eckey"], $openssl_eckeys)) {
+						array_push($input_errors, gettext("Please select a valid EC Key."));
```

Comment:
Make sure this is changed to match the label.

Reply:
fixed